.circleci/config.yml

```diff
@@ -1,53 +1,171 @@
-jobs:
-  build_and_test:
-    docker:
-      - image: 'ubuntu:bionic'
+version: 2.1
+
+commands: 
+  setup-executor:
     steps:
       - run:
-          name: installations
-          command: apt-get -qq update; apt-get install -y mysql-server ca-certificates wget build-essential git python-pip
+          name: Setup executor
+          command: |
+            apt-get -qq update
+            apt-get -q install -y git openssh-client curl ca-certificates make tar gzip
+            bash <(curl -fsSL https://get.docker.com)
+      - setup_remote_docker:
+          docker_layer_caching: true
+
+  setup-automation:
+    steps:
       - run:
-          name: Redis_5_upgrade
-          command: cd ..; git clone https://github.com/antirez/redis.git; cd ./redis; git fetch; git checkout 6.0.1; make; make install
+          name: Setup Automation
+          command: |
+            mkdir -p deps
+            (cd deps; git clone https://github.com/RedisLabsModules/readies.git)
+            ./deps/readies/bin/getpy3
+            ./sbin/system-setup.py
+
+  setup-gears:
+    steps:
       - run:
-          name: download_RedisGears
-          command: cd ..; wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears.linux-bionic-x64.master.zip; apt-get install -y unzip; unzip redisgears.linux-bionic-x64.master.zip
+          name: Setup Gears
+          command: |
+            cd ..
+            wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears.linux-bionic-x64.master.zip
+            unzip redisgears.linux-bionic-x64.master.zip
+            wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears-dependencies.linux-bionic-x64.master.tgz
+            mkdir -p /var/opt/redislabs/modules/rg
+            cd /var/opt/redislabs/modules/rg
+            tar -xvf /root/redisgears-dependencies.linux-bionic-x64.master.tgz
+
+  platform-build-steps:
+    parameters:
+      platform:
+        type: string
+    steps:
+      - setup-executor
+      - checkout
+      - setup-automation
       - run:
-          name: download_RedisGears_deps
-          command: cd ..; wget http://redismodules.s3.amazonaws.com/redisgears/snapshots/redisgears-dependencies.linux-bionic-x64.master.tgz; mkdir -p /var/opt/redislabs/modules/rg/; cd /var/opt/redislabs/modules/rg/; tar -xvf /root/redisgears-dependencies.linux-bionic-x64.master.tgz
+          name: Build for platform
+          command: |
+            GEARS_VERSION=master OSNICK=<<parameters.platform>> ./build/reqpacks/build
```

Comment:
So we will have to change the version here when we get it into a branch right?

Reply:
We also define `GEARS_VERSION` inside `build/reqoacks/build`, and I always prefer to have logic in scripts one can run outside CircleCI rather than in CircleCI itself.